strawberryfields/devicespecs/fock.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Fock backend validation data"""
+from .device_specs import DeviceSpecs
+
+
+class FockSpecs(DeviceSpecs):
+    """Validation data for the Fock backend"""
+
+    modes = 0
+    remote = True
+    interactive = True
+
+    primitives = {
+        # meta operations
+        "All",
+        "New_modes",
+        "Delete",
+        # state preparations
+        "Vacuum",
+        "Coherent",
+        "Squeezed",
+        "DisplacedSqueezed",
+        "Thermal",
+        "Fock",
+        "Catstate",
+        "Ket",
+        "DensityMatrix",
+        # measurements
+        "MeasureFock",
+        "MeasureHomodyne",
+        # channels
+        "LossChannel",
+        # single mode gates
+        "Dgate",
+        "Xgate",
+        "Zgate",
+        "Sgate",
+        "Rgate",
+        "Vgate",
+        "Kgate",
+        "Fouriergate",
+        "BSgate",
+        "CKgate",
+    }
+
+    decompositions = {
+        "Interferometer": {},
```

Comment:
Can you remind me what effect a blank dictionary has? Does that mean it supports any possible decomposition the frontend might try?

Reply:
Essentially, yes. So for instance, once #87 is merged, the `chip0` device will look like this:
```python
   decompositions = {
        "Interferometer": {"geometry": "mach_zehnder"},
   }
```
specifying that compilation of the interferometer will use the `mach_zehnder` decomposition function, as opposed to `rectangular` or `triangular`.